module/VuFind/tests/unit-tests/src/VuFindTest/Recommend/LibGuidesProfileTest.php

```diff
@@ -0,0 +1,185 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace VuFindTest\Recommend;
+
+use Laminas\Cache\Storage\Adapter\AbstractAdapter as CacheAdapter;
+use Laminas\Config\Config;
+use VuFind\Config\PluginManager as ConfigPluginManager;
+use VuFind\Connection\LibGuides;
+use VuFind\Recommend\LibGuidesProfile;
+use VuFind\Search\Base\Options;
+use VuFind\Search\Base\Params;
+use VuFindTest\Search\TestHarness\Results;
+
+/**
+ * LibGuidesProfile recommendation module Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class LibGuidesProfileTest extends \PHPUnit\Framework\TestCase
+{
+    use \VuFindTest\Feature\FixtureTrait;
+
+    /**
+     * LibGuides connection object
+     *
+     * @var LibGuides
+     */
+    protected $connector;
+
+    /**
+     * Config object
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Cache adapter object
+     *
+     * @var CacheAdapter
+     */
+    protected $cacheAdapter;
+
+    /**
+     * LibGuidesProfile object
+     *
+     * @var LibGuidesProfile
+     */
+    protected $libGuidesProfile;
+
+    /**
+     * Set up mocks for tests.
+     *
+     * @return void
+     */
+    public function setUp(): void
+    {
+        // Mock LibGuides connector
+        $this->connector = $this->getMockBuilder(LibGuides::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $accountsFixture = $this->getFixture("libguides/api/accounts");
+        $accounts = json_decode(substr($accountsFixture, strpos($accountsFixture, '[')));
+        $this->connector->method('getAccounts')->willReturn($accounts);
+
+        // Mock config and caching logic in LibGuidesProfile.
+        // Caching is from a trait, which is not the point of this test suite,
+        // and the config is only used in LibGuidesProfile for caching.
+        $this->config = $this->getMockBuilder(Config::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->cacheAdapter = $this->getMockBuilder(CacheAdapter::class)->getMock();
+
+        // For the target class LibGuidesProfile, only mock the caching methods
```

Comment:
It's probably unusual to mock (some of) the very class we're testing, but I think the reasoning is sound?

Reply:
It's actually not uncommon to do this sort of thing, especially for classes that interact with the file system or database. In this particular case, there is a "Memory" cache adapter that stores data in-memory and is good for testing, so you _could_ use that instead of mocking the cache methods... but this is fine as is. I'll merge this in its current form, and if you feel like changing it in the future, we can always do a follow-up PR.